### ISSUE TITLE
RCS1061 not triggered when comment in between

### ISSUE THREAD

**Issue by Danghor** [JosefPihrt/Roslynator#535]
**Product and Version Used**: 
Roslynator 2019 [Preview]
2.1.0

**Steps to Reproduce**:

*RCS1061: Merge if statement with nested if statement* is not triggered for the following code:

```csharp
if (result == DialogResult.OK)
{
    // Foobar
    if (openFileDialog != null) // RCS1061 not triggered
    {

    }
}
```

**Actual Behavior**:
No diagnostic triggered.

**Expected Behavior**:
The diagnostic should be triggered and the code fix should move the comment to before the if-statement:

```csharp
// Foobar
if (result == DialogResult.OK && openFileDialog != null)
{

}
```

However, maybe this is also intended this way. What do you think @JosefPihrt ?

**Comment by JosefPihrt** [comment]
Yes, this is by design.
